cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/TapChangerEq.java

```diff
@@ -65,14 +72,25 @@ public static void writePhaseTablePoint(String id, String phaseTapChangerTableId
         writer.writeEndElement();
     }
 
-    public static void writeRatio(String id, String tapChangerName, String transformerEndId, double lowStep, double highStep, double neutralStep, double normalStep, double neutralU, boolean ltcFlag, double stepVoltageIncrement, String ratioTapChangerTableId, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
+    public static void writeRatio(String id, String tapChangerName, String transformerEndId, double lowStep, double highStep, double neutralStep, double normalStep, double neutralU, boolean ltcFlag, double stepVoltageIncrement, String ratioTapChangerTableId, Optional<String> regulatingControlId, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
         CgmesExportUtil.writeStartIdName("RatioTapChanger", id, tapChangerName, cimNamespace, writer);
         CgmesExportUtil.writeReference(EQ_RATIOTAPCHANGER_TRANSFORMEREND, transformerEndId, cimNamespace, writer);
         writer.writeStartElement(cimNamespace, EQ_RATIOTAPCHANGER_SVI);
         writer.writeCharacters(CgmesExportUtil.format(stepVoltageIncrement));
         writer.writeEndElement();
         writeSteps(lowStep, highStep, neutralStep, normalStep, neutralU, ltcFlag, cimNamespace, writer);
         CgmesExportUtil.writeReference(EQ_RATIOTAPCHANGER_RATIOTAPCHANGERTABLE, ratioTapChangerTableId, cimNamespace, writer);
+        if (regulatingControlId.isPresent()) {
+            CgmesExportUtil.writeReference("TapChanger.TapChangerControl", regulatingControlId.get(), cimNamespace, writer);
+        }
+        writer.writeEndElement();
+    }
+
+    public static void writeControl(String id, String name, String mode, String terminalId, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
+        CgmesExportUtil.writeStartIdName("TapChangerControl", id, name, cimNamespace, writer);
+        writer.writeEmptyElement(cimNamespace, "RegulatingControl.mode");
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, String.format("%s%s.%s", cimNamespace, "RegulatingControlModeKind", mode));
```

Comment:
Here, `mode` is always written but it can be null

Reply:
`only` valid regulating modes will be written. when a tap changer has IIDM regulation mode FIXED_TAP it won't be exported as a regulating control to CGMES.